### PR TITLE
Avijit/add "include" directory to whl and few enhancements

### PR DESCRIPTION
**Pull request by avijit-nervana**
* Added support for including nGraph include files to the Python Whl so that other backends can be built with the binary distribution of `ngraph_tensorflow_bridge`.
* Modified the Python format checking/applying code
* Added API: `ngraph_tf_cxx11_abi_flag()` that returns the value of the `_GLIBCXX_USE_CXX11_ABI` used building the bridge and nGraph library. This will be useful for building other C++ libraries using the bridge (such as a backend).

### CODE REVIEW

File: build_ngtf.py [hunk 1]
```diff
@@ -132,8 +146,8 @@ def main():
 
     # The cxx_abi flag is translated to _GLIBCXX_USE_CXX11_ABI
     # For gcc 4.8 - this flag is set to 0 and newer ones, this is set to 1
-    # The scpeific value is determined from the TensorFlow build 
-    # Normally the shipped TensorFlow is built with gcc 4.8 and thus this 
+    # The scpeific value is determined from the TensorFlow build
```
Comment (sayantan-nervana) [inline build_ngtf.py]:
```suggestion
    # The specific value is determined from the TensorFlow build
```

File: maint/apply-code-format.sh [hunk 2]
```diff
@@ -36,24 +36,8 @@ else
...
-    bash_lib_print_error "Failed invocation of Python."
-    exit 1
-fi
-
-
 declare CLANG_FORMAT_BASENAME="clang-format-3.9"
 declare REQUIRED_CLANG_FORMAT_VERSION=3.9
-if [[ "3" == "${PYTHON_VERSION}" ]]; then
-    declare YAPF_FORMAT_BASENAME="yapf"
-    declare REQUIRED_YAPF_FORMAT_VERSION=0.26
```
Comment (sayantan-nervana) [inline maint/apply-code-format.sh]: right now its not asserted if its yapf 0.26 or some other version. Which means that everytime yapf releases a new version we might have to commit a new change that reformats python code according to the latest version of yapf. Which is not a bad idea: then the idea is that our repo is always formatted by latest yapf.

in that case, we need to make changes in lines like this:
https://github.com/NervanaSystems/ngraph-tf/blob/dc93c49be6a98e0f149c914c25f0280e88f3488f/test/ci/docker/docker-scripts/run-check-code-format.sh#L129
https://github.com/NervanaSystems/ngraph-tf/blob/dc93c49be6a98e0f149c914c25f0280e88f3488f/test/ci/docker/dockerfiles/Dockerfile.ngraph-tf-ci-py2#L63
https://github.com/NervanaSystems/ngraph-tf/blob/dc93c49be6a98e0f149c914c25f0280e88f3488f/test/ci/docker/dockerfiles/Dockerfile.ngraph-tf-ci-py3#L67

File: maint/check-code-format.sh [hunk 3]
```diff
@@ -37,23 +37,10 @@ else
...
-fi
-
 declare CLANG_FORMAT_BASENAME="clang-format-3.9"
 declare REQUIRED_CLANG_FORMAT_VERSION=3.9
-if [[ "3" == "${PYTHON_VERSION}" ]]; then
-    declare YAPF_FORMAT_BASENAME="yapf"
-    declare REQUIRED_YAPF_FORMAT_VERSION=0.26
-fi
+declare YAPF_FORMAT_BASENAME="yapf"
+declare REQUIRED_YAPF_FORMAT_VERSION=0.26
```
Comment (sayantan-nervana) [inline maint/check-code-format.sh]: the required version is declared but not enforced/checked

File: maint/apply-code-format.sh (hunk 2 above)
Comment (avijit-nervana) [reply maint/apply-code-format.sh]: Agree. We will need to add that in a separate PR.

File: maint/check-code-format.sh (hunk 3 above)
Comment (avijit-nervana) [reply maint/check-code-format.sh]: Done.

File: maint/apply-code-format.sh [hunk 4]
```diff
@@ -36,25 +36,23 @@ else
...
+declare YAPF_FORMAT_BASENAME="yapf"
+declare REQUIRED_YAPF_FORMAT_VERSION=0.26.0
+
+# Check the YAPH format
+declare YAPH_VERSION=`python -c "import yapf; print(yapf.__version__)"`
```
Comment (sayantan-nervana) [inline maint/apply-code-format.sh]:
```suggestion
declare YAPF_VERSION=`python -c "import yapf; print(yapf.__version__)"`
```

File: maint/apply-code-format.sh [hunk 5]
```diff
@@ -36,25 +36,23 @@ else
...
+
+if [[ "${YAPH_VERSION}" != "${REQUIRED_YAPF_FORMAT_VERSION}" ]] ; then
```
Comment (sayantan-nervana) [inline maint/apply-code-format.sh]:
```suggestion
if [[ "${YAPF_VERSION}" != "${REQUIRED_YAPF_FORMAT_VERSION}" ]] ; then
```

File: maint/apply-code-format.sh [hunk 6]
```diff
@@ -36,25 +36,23 @@ else
...
+    echo -n "Unable to match version for ${YAPF_FORMAT_BASENAME}"
+    echo -n " Required: ${REQUIRED_YAPF_FORMAT_VERSION}"
+    echo  " Installed: ${YAPH_VERSION}"
```
Comment (sayantan-nervana) [inline maint/apply-code-format.sh]:
```suggestion
    echo  " Installed: ${YAPF_VERSION}"
```

File: maint/check-code-format.sh [hunk 7]
```diff
@@ -37,22 +37,19 @@ else
...
 declare CLANG_FORMAT_BASENAME="clang-format-3.9"
 declare REQUIRED_CLANG_FORMAT_VERSION=3.9
-if [[ "3" == "${PYTHON_VERSION}" ]]; then
-    declare YAPF_FORMAT_BASENAME="yapf"
-    declare REQUIRED_YAPF_FORMAT_VERSION=0.26
+declare YAPF_FORMAT_BASENAME="yapf"
+declare REQUIRED_YAPF_FORMAT_VERSION=0.26.0
+
+# Check the YAPH format
+declare YAPH_VERSION=`python -c "import yapf; print(yapf.__version__)"`
```
Comment (sayantan-nervana) [inline maint/check-code-format.sh]:
```suggestion
declare YAPF_VERSION=`python -c "import yapf; print(yapf.__version__)"`
```

File: maint/check-code-format.sh [hunk 8]
```diff
@@ -37,22 +37,19 @@ else
...
+
+if [[ "${YAPH_VERSION}" != "${REQUIRED_YAPF_FORMAT_VERSION}" ]] ; then
```
Comment (sayantan-nervana) [inline maint/check-code-format.sh]:
```suggestion
if [[ "${YAPF_VERSION}" != "${REQUIRED_YAPF_FORMAT_VERSION}" ]] ; then
```

File: maint/check-code-format.sh [hunk 9]
```diff
@@ -37,22 +37,19 @@ else
...
+    echo -n "Unable to match version for ${YAPF_FORMAT_BASENAME}"
+    echo -n " Required: ${REQUIRED_YAPF_FORMAT_VERSION}"
+    echo  " Installed: ${YAPH_VERSION}"
```
Comment (sayantan-nervana) [inline maint/check-code-format.sh]:
```suggestion
    echo  " Installed: ${YAPF_VERSION}"
```

File: maint/apply-code-format.sh [hunk 10]
```diff
@@ -36,25 +36,23 @@ else
...
-
 declare CLANG_FORMAT_BASENAME="clang-format-3.9"
 declare REQUIRED_CLANG_FORMAT_VERSION=3.9
-if [[ "3" == "${PYTHON_VERSION}" ]]; then
-    declare YAPF_FORMAT_BASENAME="yapf"
-    declare REQUIRED_YAPF_FORMAT_VERSION=0.26
+declare YAPF_FORMAT_BASENAME="yapf"
+declare REQUIRED_YAPF_FORMAT_VERSION=0.26.0
+
+# Check the YAPH format
```
Comment (sayantan-nervana) [inline maint/apply-code-format.sh]:
```suggestion
# Check the YAPF format
```

File: maint/check-code-format.sh [hunk 11]
```diff
@@ -37,22 +37,19 @@ else
...
-
 declare CLANG_FORMAT_BASENAME="clang-format-3.9"
 declare REQUIRED_CLANG_FORMAT_VERSION=3.9
-if [[ "3" == "${PYTHON_VERSION}" ]]; then
-    declare YAPF_FORMAT_BASENAME="yapf"
-    declare REQUIRED_YAPF_FORMAT_VERSION=0.26
+declare YAPF_FORMAT_BASENAME="yapf"
+declare REQUIRED_YAPF_FORMAT_VERSION=0.26.0
+
+# Check the YAPH format
```
Comment (sayantan-nervana) [inline maint/check-code-format.sh]:
```suggestion
# Check the YAPF format
```

**Pull request merged**